### ISSUE TITLE
Unable to add new employee

### ISSUE THREAD

**Issue by rlwy** [sapplica/sentrifugo#27]
I've got a fresh install of Sentrifugo in my local computer for testing. My installation went fine and I managed to setup the basics. However, when I filled in the form to add a new employee, upon clicking the "Save" button, the loading spinner appears and then disappears but nothing happens. I am still on the "Add New Employee" page and my form has not been submitted, no new employee was created. The log is empty and there are no error messages appearing.
<img width="1268" alt="screen shot 2016-04-19 at 12 50 53 am" src="https://cloud.githubusercontent.com/assets/7339316/14612279/d900640c-05c8-11e6-8947-a5853de4612c.png">

**Comment by heinnge** [comment]
Not sure we have the exact same problem. but I'm also getting "something went wrong" error even though all the required fields are correctly filled up.

![Same Problem](https://cloud.githubusercontent.com/assets/14687/14938232/220dd7aa-0f43-11e6-8c14-9e6f4754374d.JPG)

**Comment by manuel-lookatitude** [comment]
Could this be the same as #28 ?

**Comment by sapplica** [comment]
Hi,

Which version are you using? Did you install the application from scratch or have you 
upgraded it? Kindly check if you have the table 'main_emailLogs' in your database.

**Comment by mpsoares13** [comment]
I have a fix for this and would like to make a pull request. It's about validating non mandatory fields.

**Comment by heinnge** [comment]
yes, i think i'm having the same problem as @mpsoares13 i look up the code and found the same where the logic try to validate the optional fields.

**Comment by prakashrgp** [comment]
@mpsoares13 @heinnge Hi guys, I am also facing this same issue. Can you guys help me out in fixing this. Please describe about the fix. It would be very helpful for me.

**Comment by mpsoares13** [comment]
@prakashrgp 

You need to edit the save method in the EmployeeController, starting from this point: https://github.com/sapplica/sentrifugo/blob/master/application/modules/default/controllers/EmployeeController.php#L1571

Simple solution, replace lines like:
`'prefix_id'=>$prefix_id,`
with
`'prefix_id'=>isset($prefix_id) ? $prefix_id : null,`
And to this for all the non required fields.

If you really want to understand what's happening, change the try - catch and after the rollback statement (https://github.com/sapplica/sentrifugo/blob/master/application/modules/default/controllers/EmployeeController.php#L1760) debug the exception to know what's the missing field. If you don't want to bother with a debugger just do something like:
`var_dump($e);die();`

**Comment by prakashrgp** [comment]
hi @mpsoares13 @heinnge  i couldn't get it working even after those changes mentioned above. when i try to submit the management details the spinner loads and disappears and nothing is happening after that and i couldn't save the details. 

![image](https://cloud.githubusercontent.com/assets/19578535/15565677/dffbfa3a-2339-11e6-90a3-ae09088b880b.png)

Could you guys help me further. I have attached my EmployeeController.php file here. [EmployeeController.zip](https://github.com/sapplica/sentrifugo/files/283575/EmployeeController.zip) Please have a look at it and if you could please make the changes. I have a demo for this application tomorrow and i am stuck at this now. Any help would be really a great one for me. Thanks

**Comment by lgbmontenegroii** [comment]
Hi all. I am also encountering the same issue. The WebApp is returning the error message 'Something went wrong'. I'm using version 2.1.1 and it's a fresh install. Did not skip any steps during the configuration phase.

**Comment by Shanmukha0015** [comment]
Hi all, I am a newbie to Sentrifugo. Recently I had installed Sentrifugo and configured it. But when I am trying to add a new employee the spinner rotates when I click the add button but doesn't create the new employee.
Here are
![issue1](https://cloud.githubusercontent.com/assets/16813666/17129226/7b84946a-532f-11e6-940c-837ddc210f11.PNG)
![issue2](https://cloud.githubusercontent.com/assets/16813666/17129227/7b87a1dc-532f-11e6-9618-ce8d111b8d14.png)

 the attached screenshots of my application. 
Please do suggest me how to add the employee coz I am stuck at this point.Any help is appreciated.

**Comment by mvillard** [comment]
Hi team,

Same error here. Unable to add any employee (even the first manager).
No logs available in logs/.
Clicking on 'SAVE' form just reload the URL/employee page.

Thank you for your help!

**Comment by mohamed--abdel-maksoud** [comment]
I ran into this problem when I migrated sentrifugo to a different database and user. The solution is to make sure the triggers are defined by the current database user. One this is fixed, things work like a charm.

Before I complain I'll express my sincere gratitude to sapplica for this great software. The application seriously needs to log exceptions though, otherwise we are left in the dark when something like this happens.

**Comment by mvillard** [comment]
Hi Mohamed,

Thanks for your feedback. I have done this check already and this is already the right setup.

Thanks,

**Comment by ajaytripathi** [comment]
Unable to add new employee on fresh install on Ubuntu. DB triggers are applied and exist.

I am able to add External Users, But not Employee.
> SHOW TRIGGERS LIKE '%employees%';
2 rows in set (0.01 sec)

>SHOW TRIGGERS;
36 rows in set (0.01 sec)

mysql> SHOW GRANTS FOR 'testorg';
+----------------------------------------------------------------------------+
| Grants for gridstore@%                                                     |
+----------------------------------------------------------------------------+
| GRANT USAGE ON *.* TO 'testorg'@'%'                                      |
| GRANT ALL PRIVILEGES ON `testorg`.* TO 'testorg'@'%' WITH GRANT OPTION |
+----------------------------------------------------------------------------+
2 rows in set (0.00 sec)

**Comment by ajaytripathi** [comment]
Solved the problem I was facing - default value of "date of leaving' is 000-00-00, under mysql strict mode this is not  allowed. One has to run mysql in no-strict mode 

Change or add following in file /etc/mysql/conf.d/mysql.cnf should solve the issue. 
[mysqld]
sql_mode=NO_ENGINE_SUBSTITUTION

I hope this will get me starting ...

**Comment by bigaltheking** [comment]
yes, the strict mode was causing me issues as well. Is this expected?

**Comment by dragancla** [comment]
Thanks, @ajaytripathi ! That also did it for me!

For those with wamp, the location of the file is ..\wamp\bin\mysql\mysql5.7.14\my.ini
Just add 
`sql-mode="NO_ENGINE_SUBSTITUTION"`
somewhere there and then restart the mysql service and refresh the Employees page.

**Comment by princekwes** [comment]
The message i get:
First name cannot be empty at row 8.

during the import of employees with excel import file.

if for example i create 12 format employees to import i get the error that the next one which is not part of the excel file cant have first name empty.

Example i have ready an import file with all columns containing 12 names and other info.

Then i get the error:
First name cannot be empty at row 13.

Please help me asap

**Comment by mikecruz302** [comment]
I have the same issue as above First name cannot be empty at row 8. Can anyone help

**Comment by sinaghazi** [comment]
There is a workaround for this problem: 
On your mysql table "main_employees" set the default value of "date_of_leaving" to some future date.

**Comment by mikecruz302** [comment]
That did not resolve the issue. Still occurs


[cid:34d6299c-ec74-4622-affd-8c439a5c56cc][cid:d0df4ad6-e305-452c-ae26-f362ca6d0e59]?

Sincerely,

Mike Cruz
The Pension Studio
1226 Omar Rd
West Palm Beach, FL  33405
Office: (561) 459-4871 x 129
Cell: (561) 666-0244
Fax: (800) 883-1970
Mike@thepensionstudio.com<mailto:%20mike@thepensionstudio.com>

....Creating the future you envision...

CONFIDENTIALITY NOTICE: This communication is confidential, may be privileged and is meant only for the intended recipient.  If you are not the intended recipient, please notify the sender ASAP and delete this message from your system.

________________________________

**Comment by sinaghazi** [comment]
on version 3.2 this issue of adding the first user (organization head) appeared again and it gets resolved by: 

ALTER TABLE `main_employees` CHANGE `date_of_joining` `date_of_joining` DATE NULL DEFAULT '1970-01-01', CHANGE `date_of_leaving` `date_of_leaving` DATE NULL DEFAULT '2020-01-01'; 

I didn't check for the other issue posted in comments.

**Comment by mikecruz302** [comment]
Oh I see I dont have that issue mine is I cant import I keep getting

First name cannot be empty at row 41.
?


Sincerely,

Mike Cruz
The Pension Studio
1226 Omar Rd
West Palm Beach, FL  33405
Office: (561) 459-4871 x 129
Cell: (561) 666-0244
Fax: (800) 883-1970
Mike@thepensionstudio.com<mailto:%20mike@thepensionstudio.com>

....Creating the future you envision...

CONFIDENTIALITY NOTICE: This communication is confidential, may be privileged and is meant only for the intended recipient.  If you are not the intended recipient, please notify the sender ASAP and delete this message from your system.

________________________________

**Comment by fauzihafsar** [comment]
hi
good morning, I have questions like this

I just installed it and when I tried adding new employees, but I can't add it,
and when I want to see a business unit but what appears is like this,
so is the background check that displays the same display as the business unit in the employee / candidate screening section
please help me in solving this problem, thank you very much
![hr](https://user-images.githubusercontent.com/34773476/54003554-2f357880-4185-11e9-9b2e-aa34d4c8413f.png)
![business_unit](https://user-images.githubusercontent.com/34773476/54003555-2f357880-4185-11e9-93eb-be2f43cbd8e1.png)
![bg_chek](https://user-images.githubusercontent.com/34773476/54003556-2fce0f00-4185-11e9-879a-86e1be8ccd92.png)

**Comment by excedra** [comment]
> on version 3.2 this issue of adding the first user (organization head) appeared again and it gets resolved by:
> 
> ALTER TABLE `main_employees` CHANGE `date_of_joining` `date_of_joining` DATE NULL DEFAULT '1970-01-01', CHANGE `date_of_leaving` `date_of_leaving` DATE NULL DEFAULT '2020-01-01';
> 
> I didn't check for the other issue posted in comments.

Done a fresh install, had issue that i cannot add the first employee. 
Done your SQL, works fine. now. Fixed (for now?)
